code/datums/components/manual_heart.dm

```diff
@@ -0,0 +1,171 @@
+/**
+ * Manual heart pumping component. Requires the holder to pump their heart manually every
+ * so often or die.
+ *
+ * Mainly used by the cursed heart.
+ */
+/datum/component/manual_heart
+	/// The action for pumping your heart
+	var/datum/action/cooldown/manual_heart/pump_action
+	var/last_pump = 0
+	var/add_colour = TRUE //So we're not constantly recreating colour datums
+	/// How long between needed pumps; you can pump one second early
+	var/pump_delay = 3 SECONDS
+	/// How much blood volume you lose every missed pump, this is a flat amount not a percentage!
+	var/blood_loss = (BLOOD_VOLUME_NORMAL / 5) // 20% of normal volume, missing five pumps is instant death
```

Comment:
```suggestion
	var/blood_loss = BLOOD_VOLUME_NORMAL * 0.2 // 20% of normal volume, missing five pumps is instant death
```
something something multiplication better
multiplication does not need brackets. can be cleaned up further in the file